GitHubAction/GIT/GITInfo.cs

```diff
@@ -0,0 +1,101 @@
+﻿namespace GIT
+{
+    using System.IO;
+    using System.Management.Automation;
+
+    public class GITInfo : IGITInfo
+    {
+        public GITInfo()
+        {
+            AllowWritesOnDirectory(Directory.GetCurrentDirectory());
+
+            using (PowerShell powershell = PowerShell.Create())
+            {
+                powershell.AddScript("git --version");
+                var gitVersion = powershell.Invoke().FirstOrDefault()?.ToString();
+                powershell.Commands.Clear();
+
+                powershell.AddScript($"git config --global --add safe.directory {Directory.GetCurrentDirectory()}");
+                powershell.Invoke();
+                powershell.Commands.Clear();
+
+                powershell.AddScript($"git fetch --all --tags --force --quiet");
+                powershell.Invoke();
+                powershell.Commands.Clear();
+
+                if (powershell.HadErrors)
+                {
+                    powershell.AddScript("$PSVersionTable.PSVersion");
+                    var version = String.Join(",", powershell.Invoke());
+                    powershell.Commands.Clear();
+
+                    powershell.AddScript("$PSVersionTable.Platform");
+                    var checkOS = String.Join(",", powershell.Invoke());
+                    powershell.Commands.Clear();
+
+                    string resultString = "errors: " + String.Join(",", powershell.Streams.Error.ReadAll());
+                    throw new InvalidOperationException("GIT Initial Setup Failed with PowerShell Errors: " + resultString + Environment.NewLine + " ps version: " + version + " OS: " + checkOS + " git version:" + gitVersion + "--end");
```

Comment:
use interpolated string?

Reply:
Yes. Still got 'old-school' string in my fingers when I don't pay attention.
Is the below ok? Not sure about Environment.NewLine

```suggestion
                    throw new InvalidOperationException($"GIT Initial Setup Failed with PowerShell Errors: {resultString} {Environment.NewLine} ps version: {version} OS: {checkOS} git version: {gitVersion} --end");
```